pkg/controllermanager/controller/project/project_control_reconcile.go

```diff
@@ -251,6 +251,12 @@ func (c *defaultControl) reconcileNamespaceForProject(ctx context.Context, garde
 		ns.Labels = utils.MergeStringMaps(ns.Labels, projectLabels)
 		ns.Annotations = utils.MergeStringMaps(ns.Annotations, projectAnnotations)
 
+		// If the project is reconciled for the first time then its observed generation is 0. Only in this case we want
+		// to add the "keep-after-project-deletion" annotation to the namespace when we adopt it.
+		if project.Status.ObservedGeneration == 0 {
+			ns.Annotations = utils.MergeStringMaps(ns.Annotations, map[string]string{common.NamespaceKeepAfterProjectDeletion: "true"})
```

Comment:
Can this be simplified? Or what am I missing?
```suggestion
			ns.Annotations[common.NamespaceKeepAfterProjectDeletion] = "true"
```

Reply:
done